src/notifications/ObservablePayment.java

```diff
@@ -0,0 +1,46 @@
+package notifications;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import shoppingCart.ShoppingCart;
+
+/*
+ * This class is a copy of the 'Payment' class in package 'shoppingCart' that implements the Subject interface 
+ * */
+public abstract class ObservablePayment implements Subject {
+	static int nextTransactionNumber = 1;

```

Comment:
You'll need to provide a thread safe counter. You can check syncronized methods, AtomicInteger...